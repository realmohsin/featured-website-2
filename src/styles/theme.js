import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
