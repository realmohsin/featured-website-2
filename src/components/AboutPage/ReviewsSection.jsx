import React, { useState } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { Grid, Container } from '@material-ui/core'
import Review from './Review'
import Button from '../Button'
import reviews from '../../data/reviews'

const useStyles = makeStyles(theme => ({
  teamSection: {
    padding: '6rem 0 12rem'
  },
  teamSectionTitle: {
    ...theme.custom.title2,
    textAlign: 'left',
    fontSize: '5.6rem',
    paddingLeft: '5rem',
    marginBottom: '1rem',
    [theme.breakpoints.down('md')]: {
      fontSize: '4.6rem'
    },
    [theme.breakpoints.down('xs')]: {
      fontSize: '3.4rem',
      paddingLeft: '3.2rem',
      marginBottom: 0
    }
  },
  moreReviewsButton: {
    margin: '3rem auto'
  }
}))

const ReviewsSection = () => {
  const classes = useStyles()

  const [moreOpen, setMoreOpen] = useState(false)

  const toggleOpenMore = e => {
    setMoreOpen(moreOpen => !moreOpen)
  }

  return (
    <section className={classes.teamSection} id='reviews-section'>
      <Container>
        <h2 className={classes.teamSectionTitle}>Reviews</h2>
        <Grid container>
          {reviews.slice(0, 6).map(review => (
            <Review key={review.id} review={review} />
          ))}
          {moreOpen &&
            reviews.slice(5, 11).map(review => <Review key={review.id} review={review} />)}
        </Grid>
        <Button
          handleClick={toggleOpenMore}
          extraClassName={classes.moreReviewsButton}
          text={`See ${moreOpen ? 'Less' : 'More'} Reviews`}
        />
      </Container>
    </section>
  )
}

export default ReviewsSection
