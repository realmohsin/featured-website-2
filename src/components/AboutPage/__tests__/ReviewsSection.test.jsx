import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import ReviewsSection from "../ReviewsSection";

// Mock Review component
jest.mock("../Review", () => {
  return function MockReview({ review }) {
    return (
      <div data-testid={`review-${review.id || review.name}`}>
        <span>{review.name}</span>
        <span>{review.content}</span>
      </div>
    );
  };
});

// Mock Button component
jest.mock("../../Button", () => {
  return function MockButton({ text, handleClick, extraClassName }) {
    return (
      <button onClick={handleClick} className={extraClassName}>
        {text}
      </button>
    );
  };
});

// Mock reviews data
jest.mock("../../../data/reviews", () => [
  { id: 1, name: "John Doe", content: "Great service!" },
  { id: 2, name: "Jane Smith", content: "Excellent lawyers!" },
  { id: 3, name: "Bob Johnson", content: "Highly recommend!" },
  { id: 4, name: "Alice Brown", content: "Professional team!" },
  { id: 5, name: "Charlie Wilson", content: "Outstanding work!" },
  { id: 6, name: "Diana Davis", content: "Very satisfied!" },
  { id: 7, name: "Eve Miller", content: "Top quality service!" },
  { id: 8, name: "Frank Garcia", content: "Exceeded expectations!" },
  { id: 9, name: "Grace Lee", content: "Fantastic experience!" },
  { id: 10, name: "Henry Taylor", content: "Would hire again!" },
  { id: 11, name: "Ivy Anderson", content: "Best law firm!" },
]);

const theme = createMuiTheme({
  custom: {
    title2: {
      fontSize: "5.6rem",
      fontWeight: "bold",
    },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("ReviewsSection Component", () => {
  it("renders reviews section with initial reviews", () => {
    const { container } = renderWithTheme(<ReviewsSection />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("displays section title", () => {
    renderWithTheme(<ReviewsSection />);
    expect(screen.getByRole("heading", { name: /reviews/i })).toBeInTheDocument();
  });

  it("renders first 6 reviews initially", () => {
    renderWithTheme(<ReviewsSection />);
    
    expect(screen.getByTestId("review-1")).toBeInTheDocument();
    expect(screen.getByTestId("review-2")).toBeInTheDocument();
    expect(screen.getByTestId("review-3")).toBeInTheDocument();
    expect(screen.getByTestId("review-4")).toBeInTheDocument();
    expect(screen.getByTestId("review-5")).toBeInTheDocument();
    expect(screen.getByTestId("review-6")).toBeInTheDocument();
  });

  it("does not show additional reviews initially", () => {
    renderWithTheme(<ReviewsSection />);
    
    expect(screen.queryByTestId("review-7")).not.toBeInTheDocument();
    expect(screen.queryByTestId("review-8")).not.toBeInTheDocument();
  });

  it("shows 'See More Reviews' button initially", () => {
    renderWithTheme(<ReviewsSection />);
    expect(screen.getByText(/see more reviews/i)).toBeInTheDocument();
  });

  it("expands to show more reviews when button is clicked", () => {
    renderWithTheme(<ReviewsSection />);
    
    const moreButton = screen.getByText(/see more reviews/i);
    fireEvent.click(moreButton);
    
    expect(screen.getByTestId("review-7")).toBeInTheDocument();
    expect(screen.getByTestId("review-8")).toBeInTheDocument();
    expect(screen.getByTestId("review-9")).toBeInTheDocument();
  });

  it("changes button text to 'See Less Reviews' when expanded", () => {
    renderWithTheme(<ReviewsSection />);
    
    const moreButton = screen.getByText(/see more reviews/i);
    fireEvent.click(moreButton);
    
    expect(screen.getByText(/see less reviews/i)).toBeInTheDocument();
    expect(screen.queryByText(/see more reviews/i)).not.toBeInTheDocument();
  });

  it("collapses reviews when 'See Less' is clicked", () => {
    renderWithTheme(<ReviewsSection />);
    
    // Expand first
    const moreButton = screen.getByText(/see more reviews/i);
    fireEvent.click(moreButton);
    
    // Then collapse
    const lessButton = screen.getByText(/see less reviews/i);
    fireEvent.click(lessButton);
    
    expect(screen.queryByTestId("review-7")).not.toBeInTheDocument();
    expect(screen.getByText(/see more reviews/i)).toBeInTheDocument();
  });

  it("has correct section id for navigation", () => {
    renderWithTheme(<ReviewsSection />);
    const section = document.querySelector("section");
    expect(section).toHaveAttribute("id", "reviews-section");
  });
});
