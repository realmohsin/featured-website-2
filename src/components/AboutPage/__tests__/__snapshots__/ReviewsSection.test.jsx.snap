// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsSection Component renders reviews section with initial reviews 1`] = `
<section
  class="makeStyles-teamSection-1"
  id="reviews-section"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h2
      class="makeStyles-teamSectionTitle-2"
    >
      Reviews
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        data-testid="review-1"
      >
        <span>
          John Doe
        </span>
        <span>
          Great service!
        </span>
      </div>
      <div
        data-testid="review-2"
      >
        <span>
          Jane Smith
        </span>
        <span>
          Excellent lawyers!
        </span>
      </div>
      <div
        data-testid="review-3"
      >
        <span>
          Bob Johnson
        </span>
        <span>
          Highly recommend!
        </span>
      </div>
      <div
        data-testid="review-4"
      >
        <span>
          Alice Brown
        </span>
        <span>
          Professional team!
        </span>
      </div>
      <div
        data-testid="review-5"
      >
        <span>
          Charlie Wilson
        </span>
        <span>
          Outstanding work!
        </span>
      </div>
      <div
        data-testid="review-6"
      >
        <span>
          Diana Davis
        </span>
        <span>
          Very satisfied!
        </span>
      </div>
    </div>
    <button
      class="makeStyles-moreReviewsButton-3"
    >
      See More Reviews
    </button>
  </div>
</section>
`;
